### PR TITLE
Home: top course should be most recently assigned, most recent progress

### PR DESCRIPTION
**Pull request by Erin007**
[LP-770](https://codedotorg.atlassian.net/browse/LP-770)

Elizabeth noticed that teachers and students in the CSD pilot group couldn't find the assigned CSD pilot scripts on the homepage. They were there when the View More courses button was clicked, but buried in the users' other courses/scripts. The expectation is that the pilot scripts should appear first in the courses section as a top course because they were most recently assigned. 

This highlighted a bug wherein hidden scripts were excluded from being a user's top course even if they should be visible due to the user's enrollment in a pilot experiment. 

A user's top course (more accurately "assignable" since it could be a course OR a script) is based on the logic for `primary_script`. Previously `primary_script` was the first in the collection of `working_on_scripts` for the user. `working_on_scripts` was the group of scripts where user_scripts.completed_at is null, which doesn't take experiment enrollment into account. We were also relying on the order these user_scripts were returned in the array rather than looking at timestamps. This resulted in the pilot script appearing for the user, but not as a top assignable. 

BEFORE:
Screen Shot 2019-09-25 at 10 44 45 AM

I updated the `primary_script` logic to set the top course to the most recently assigned script if the user has permission to see that script; if that doesn't exist use the script the user most recently made progress in. 
AFTER:
Screen Shot 2019-09-25 at 5 38 19 PM

**Notes about hiding (because I always have to re-remind myself)**: 

 Scripts with the pilot [experiment property](https://github.com/code-dot-org/code-dot-org/blob/1ad9501b39f1079a0546ff5ea4a104e9d2a1714f/dashboard/config/scripts/csd3-pilot.script#L11) [are hidden](https://github.com/code-dot-org/code-dot-org/blob/1ad9501b39f1079a0546ff5ea4a104e9d2a1714f/dashboard/app/models/script.rb#L72) for users who do not have [pilot access](https://github.com/code-dot-org/code-dot-org/blob/1ad9501b39f1079a0546ff5ea4a104e9d2a1714f/dashboard/app/models/script.rb#L1652) that matches the experiment name unless the user has hidden_script_access. 

This is different from hiding a script in the teacher dashboard: 
Screen Shot 2019-09-25 at 5 23 30 PM

And also different from when a teacher hides a script (or stage) for the students in their section:
Screen Shot 2019-09-25 at 5 23 11 PM

### CODE REVIEW

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -1579,10 +1579,8 @@ def recent_courses_and_scripts(exclude_primary_script)
 
     # Filter out user_scripts that are already covered by a course
     course_scripts_script_ids = courses_as_student.map(&:default_course_scripts).flatten.pluck(:script_id).uniq
-
     user_scripts = in_progress_and_completed_scripts.
       select {|user_script| !course_scripts_script_ids.include?(user_script.script_id)}
-
```
Comment (davidsbailey) [inline dashboard/app/models/user.rb:1585]: revert whitespace changes?

File: dashboard/test/models/user_test.rb [hunk 2]
```diff
@@ -1597,20 +1597,20 @@ def assert_fails_email_uniqueness_validation(user)
...
 
     # add an assigned script that's more recent
-    a = create :user_script, user: user, started_at: (Time.now - 1.day)
-    assert_equal [a.script, s2.script, s1.script], user.working_on_scripts
+    a = create :user_script, user: user, assigned_at: (Time.now - 1.day)
+    assert_equal a.script, user.most_recently_assigned_script
     assert_equal a.script, user.primary_script
 
-    # make progress on an older script
+    # make progress on a older script, assignment is still primary
```
Comment (davidsbailey) [inline dashboard/test/models/user_test.rb:1609]: "an older" is grammatically correct, so please don't change it to "a older".

I like this improvement to the comments, but I'm not totally sure I follow -- what does "assignment is still primary" mean?

**Review by davidsbailey — APPROVED** [review]
Thank you for adding tests! Just one question inline.

**Review by dmcavoy — APPROVED** [review]
Nice work Erin!

**Comment by Erin007** [comment]
@fontie715 decided we don't actually want to do this now since there are consequences for relying on assignment to determine the top course. Pilot members just have to click view more and dig for the scrip they want. 😞

**Pull request closed without merging**